ui/apps/platform/src/Containers/NetworkGraph/components/CIDRFormModal.tsx

```diff
@@ -55,14 +54,34 @@ function CIDRFormModal({ selectedClusterId, isOpen, onClose }: CIDRFormModalProp
         message: string;
     }>(emptyFormCallout);
     const [CIDRBlocksToDelete, setCIDRBlocksToDelete] = useState<string[]>([]);
+    const {
+        isCIDRFormModalOpen: isOpen,
+        toggleCIDRFormModal,
+        initialCIDRFormValue,
+        setInitialCIDRFormValue,
+    } = useCIDRFormModal();
 
     const CIDRBlockMap: Record<string, CIDRBlockEntity> = {};
     CIDRBlocks.entities?.forEach(({ entity }) => {
         CIDRBlockMap[entity.id] = entity;
     });
 
-    const initialValues =
-        CIDRBlocks.entities.length !== 0 ? CIDRBlocks : { entities: [emptyCIDRBlockRow] };
+    const initialCIDRBlockRow = initialCIDRFormValue
+        ? {
+              entity: {
+                  cidr: initialCIDRFormValue,
+                  name: '',
+                  id: '',
+              },
+          }
+        : emptyCIDRBlockRow;
+
+    const initialValues = {
+        entities:
+            CIDRBlocks.entities.length > 0
+                ? [...CIDRBlocks.entities]
+                : [...CIDRBlocks.entities, initialCIDRBlockRow],
+    };
```

Comment:
I think the logic is a little off here. I believe we want to show `initialCIDRBlockRow` if the `CIDRBlocks.entities.length === 0` or `initialCIDRFormValue` exists. Otherwise the modal won't show the user's ip search filter if there are CIDR definitions.

I also wonder if `initialCIDRFormValue` is misleading here, maybe something like `prefilledCIDRValue` would be more accurate

Reply:
Yea, you're right. My fix for the test failure might have thrown this off. Will fix 👍🏼